smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceIdentifierValidator.java

```diff
@@ -36,10 +36,24 @@ public final class ResourceIdentifierValidator extends AbstractValidator {
     @Override
     public List<ValidationEvent> validate(Model model) {
         return model.shapes(ResourceShape.class)
-                .flatMap(resource -> validateAgainstChildren(resource, model))
+                .flatMap(resource ->
+                        Stream.concat(validateAgainstChildren(resource, model),
+                                      validatePropertyRedefine(resource, model)))
                 .collect(Collectors.toList());
     }
 
+    private Stream<ValidationEvent> validatePropertyRedefine(ResourceShape resource, Model model) {
+        if (resource.hasProperties()) {
+            return resource.getProperties().keySet().stream()
```

Comment:
Should both this and [validateAgainstChildren()](https://github.com/awslabs/smithy/blob/idl-2.0/smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceIdentifierValidator.java#L43) be changed to not use `Stream<>`? Otherwise the two validation paths wouldn't join as easily to provide a joined collection of ValidationEvents.

Reply:
Sure